cpp/src/io/orc/dict_enc.cu

```diff
@@ -73,21 +73,25 @@ static __device__ void LoadNonNullIndices(volatile dictinit_state_s *s,
   for (uint32_t i = 0; i < s->chunk.num_rows; i += 512) {
     const uint32_t *valid_map = s->chunk.valid_map_base;
     uint32_t is_valid, nz_pos;
-    if (t < 16) {
+    if (t < 512 / 32) {
```

Comment:
This 512 (the block size) should be made into a constexpr and applied everywhere.